### DISCUSSION TITLE
How do I write pyinst.py?

### DISCUSSION THREAD

**Discussion by JohnDevlopment** [pyinstaller/pyinstaller#7046] [PyInstaller] [answered]
Sorry if this is a silly question, but I am new relatively new to pyinstaller...

In other projects written in Python I've seen a `pyinst.py`. Does pyinstaller have some mechanism for autogenerating this script, or do they write it themselves?

**Comment by rokm** [comment, marked as answer]
Whatever the `pyinst.py` script is and does, it is written by the project themselves.

**Reply by JohnDevlopment** [reply]
OK, thanks.

**Comment by bwoodsend** [comment]
Sometimes people like to write build scripts which run pyinstaller as a subprocess then additionally post-modify the app by codesigning it or bundling it into an msi/dmg installer. This is usually pointless since the `.spec` file that PyInstaller generates is a Python script so you can add extra steps in there. My guess would be that it's something like that.

**Reply by JohnDevlopment** [reply]
I see. I was looking into whether or not to write a build script because of a problem I ran into with pyinstaller. And I noticed that certain projects (for example, yt-dlp) will have a `pyinst.py` file. I don't know what projects are meant to use when distributing source code.
